opennms-provision/opennms-detector-jms/src/main/java/org/opennms/netmgt/provision/detector/jms/ActiveMQDetector.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.provision.detector.jms;
+
+import java.lang.reflect.UndeclaredThrowableException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import javax.jms.Connection;
+import javax.jms.Session;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.opennms.netmgt.provision.DetectRequest;
+import org.opennms.netmgt.provision.support.AgentBasedSyncAbstractDetector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>ActiveMQDetector class.</p>
+ *
+ * @author roskens
+ * @version $Id: $
+ */
+
+public class ActiveMQDetector extends AgentBasedSyncAbstractDetector<URI>{
```

Comment:
Ah, I guess I forgot to finish working on this part. I want to be able to specify in the detector a flag that lets the user flag the detector to use the nodelabel instead of the ip address.

This way, if they specify a broker URL that uses SSL, they won't get an exception if the broker's SSL certificate does not include its IP address.